### PR TITLE
test(multi-env): add EnvInfoLoader UT for skip logic and fix a bug

### PR DESCRIPTION
**Pull request by a1exwang**
- when skip is set, EnvInfo should not be loaded
- add UT for skip logic
- add a test case for NonExistEnvError to prevent the coverage drop because after the bugfix, this exception is not thrown in the happy path case.

### CODE REVIEW

File: packages/fx-core/tests/core/hooks.test.ts [hunk 1]
```diff
@@ -1668,4 +1675,110 @@ describe("Middleware", () => {
       }
     });
   });
+  describe("EnvInfoLoaderMW", () => {
```
Comment (qinezh) [inline packages/fx-core/tests/core/hooks.test.ts:1678]: how about update the UT description to distinguish the status that multi-env is enabled or not?

Comment (a1exwang) [reply packages/fx-core/tests/core/hooks.test.ts:1678]: Good point. I forgot that we also need to test cases with multenv disabled.

File: packages/fx-core/tests/core/hooks.test.ts [hunk 2]
```diff
@@ -1529,6 +1535,7 @@ describe("Middleware", () => {
 
   describe("LocalSettingsLoaderMW, ContextInjectorMW", () => {
     it("NoProjectOpenedError", async () => {
+      console.log("no project opened");
```
Comment (qinezh) [inline packages/fx-core/tests/core/hooks.test.ts:1538]: do we need to keep the log here?

**Review by qinezh — APPROVED** [review]

File: packages/fx-core/tests/core/hooks.test.ts (hunk 2 above)
Comment (a1exwang) [reply packages/fx-core/tests/core/hooks.test.ts]: removed debug code

File: packages/fx-core/tests/core/hooks.test.ts [hunk 3]
```diff
@@ -1668,4 +1674,110 @@ describe("Middleware", () => {
...
+
+      hooks(MyClass, {
+        myMethod: [MockProjectSettingsLoaderMW(), EnvInfoLoaderMW(true), SolutionContextSpyMW],
+      });
+      const my = new MyClass();
+      const res = await my.myMethod(inputs);
+      assert.isTrue(res.isOk());
+      assert(solutionContext);
+      // envInfo should be set to a default value when envInfo loading is skipped.
+      assert.isTrue(solutionContext?.envInfo.envName === environmentManager.getDefaultEnvName());
```
Comment (qinezh) [inline packages/fx-core/tests/core/hooks.test.ts:1756]: nit: assert.equal would be preferred here, as it provided more info such like expected string and the actual string when the test case failed.

File: packages/fx-core/tests/core/hooks.test.ts [hunk 4]
```diff
@@ -1668,4 +1674,110 @@ describe("Middleware", () => {
...
+    });
+
+    it("skip dynamically with inputs.ignoreEnvInfo", async () => {
```
Comment (qinezh) [inline packages/fx-core/tests/core/hooks.test.ts:1759]: another case we might need to added is `input.env` is provided.

Comment (a1exwang) [reply packages/fx-core/tests/core/hooks.test.ts:1759]: Yes. More case will be added in later PRs. As this PR main contains cases related to the bug fix

File: packages/fx-core/tests/core/hooks.test.ts (hunk 3 above)
Comment (a1exwang) [reply packages/fx-core/tests/core/hooks.test.ts]: It makes sense. Fixed

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**